From 879578d3f4dc9bc42aa433b1fb6b584564f83617 Mon Sep 17 00:00:00 2001
From: Eneas U de Queiroz <cotequeiroz@gmail.com>
Date: Wed, 21 Jul 2021 13:38:24 -0300
Subject: openwrt: don't quote $(SHELL) in Makefile.am

This allows to use SHELL="env bash" to get a controlled enviroment.

Signed-off-by: Eneas U de Queiroz <cotequeiroz@gmail.com>

--- a/Makefile.am
+++ b/Makefile.am
@@ -46,7 +46,7 @@ EXTRA_LTLIBRARIES	=
 # Using 'cd' in backquotes may print the directory name, use this instead:
 lt__cd		= CDPATH="$${ZSH_VERSION+.}$(PATH_SEPARATOR)" && cd
 
-git_version_gen = '$(SHELL)' '$(aux_dir)/git-version-gen' '--fallback' '$(VERSION)' '.tarball-version'
+git_version_gen = $(SHELL) '$(aux_dir)/git-version-gen' '--fallback' '$(VERSION)' '.tarball-version'
 rebuild = rebuild=:; revision=`$(lt__cd) $(srcdir) && $(git_version_gen) | $(SED) 's|-.*$$||'`
 
 
@@ -306,7 +306,7 @@ libtool: $(ltmain_sh) $(config_status) $
 	  if test 0 = '$(AM_DEFAULT_VERBOSITY)' && test 1 != '$(V)'; \
 	    then echo "  GEN     " $@; \
 	  else echo '$(SHELL) $(top_builddir)/config.status "$@"'; fi; \
-	  cd '$(top_builddir)' && '$(SHELL)' ./config.status '$@'; \
+	  cd '$(top_builddir)' && $(SHELL) ./config.status '$@'; \
 	fi
 
 
@@ -789,13 +789,13 @@ testsuite_deps_uninstalled = $(testsuite
 # Hook the test suite into the check rule
 check-local: $(testsuite_deps_uninstalled)
 	$(AM_V_at)$(CD_TESTDIR); \
-	CONFIG_SHELL='$(SHELL)' '$(SHELL)' "$$abs_srcdir/$(TESTSUITE)" \
+	CONFIG_SHELL=$(SHELL) $(SHELL) "$$abs_srcdir/$(TESTSUITE)" \
 	  $(TESTS_ENVIRONMENT) $(BUILDCHECK_ENVIRONMENT) $(TESTSUITEFLAGS)
 
 # Run the test suite on the *installed* tree.
 installcheck-local: $(testsuite_deps)
 	$(AM_V_at)$(CD_TESTDIR); \
-	CONFIG_SHELL='$(SHELL)' '$(SHELL)' "$$abs_srcdir/$(TESTSUITE)" \
+	CONFIG_SHELL=$(SHELL) $(SHELL) "$$abs_srcdir/$(TESTSUITE)" \
 	  $(TESTS_ENVIRONMENT) $(INSTALLCHECK_ENVIRONMENT) $(TESTSUITEFLAGS) \
 	  AUTOTEST_PATH='$(exec_prefix)/bin'
 
@@ -807,7 +807,7 @@ check-noninteractive-old:
 .PHONY: check-noninteractive-new
 check-noninteractive-new: $(testsuite_deps_uninstalled)
 	$(AM_V_at)$(CD_TESTDIR); \
-	CONFIG_SHELL='$(SHELL)' '$(SHELL)' "$$abs_srcdir/$(TESTSUITE)" \
+	CONFIG_SHELL=$(SHELL) $(SHELL) "$$abs_srcdir/$(TESTSUITE)" \
 	  $(TESTS_ENVIRONMENT) $(BUILDCHECK_ENVIRONMENT) \
 	  -k '!interactive' INNER_TESTSUITEFLAGS=',!interactive' \
 	  $(TESTSUITEFLAGS)
@@ -816,7 +816,7 @@ check-noninteractive-new: $(testsuite_de
 .PHONY: check-interactive
 check-interactive: $(testsuite_deps_uninstalled)
 	$(AM_V_at)$(CD_TESTDIR); \
-	CONFIG_SHELL='$(SHELL)' '$(SHELL)' "$$abs_srcdir/$(TESTSUITE)" \
+	CONFIG_SHELL=$(SHELL) $(SHELL) "$$abs_srcdir/$(TESTSUITE)" \
 	  $(TESTS_ENVIRONMENT) $(BUILDCHECK_ENVIRONMENT) \
 	  -k interactive -k recursive INNER_TESTSUITEFLAGS=',interactive' \
 	  $(TESTSUITEFLAGS)
@@ -828,7 +828,7 @@ check-noninteractive: check-noninteracti
 clean-local:
 	-$(CD_TESTDIR); \
 	test -f "$$abs_srcdir/$(TESTSUITE)" && \
-	    '$(SHELL)' "$$abs_srcdir/$(TESTSUITE)" --clean
+	    $(SHELL) "$$abs_srcdir/$(TESTSUITE)" --clean
 
 ## An empty target to depend on when a rule needs to always run
 ## whenever it is visited.
